api/src/main/java/org/cardanofoundation/rosetta/api/block/service/LedgerBlockServiceImpl.java

```diff
@@ -113,10 +115,13 @@ public Block findLatestBlock() {
 
   @Override
   public GenesisBlock findGenesisBlock() {
-    log.debug("About to run findGenesisBlock query");
-    return blockRepository.findGenesisBlock()
-        .map(b -> mapper.map(b, GenesisBlock.class))
-        .orElseThrow(ExceptionFactory::genesisBlockNotFound);
+    if(cachedGenesisBlock == null) {
```

Comment:
Maybe introduce caching here?
https://stackoverflow.com/a/26283080/25051587
In this case in theory we can reset caches with actuator..